source/fuel_client.lua

```diff
@@ -214,32 +242,33 @@ Citizen.CreateThread(function()
 					local stringCoords = GetEntityCoords(isNearPump)
 
 					if currentCash >= Config.JerryCanCost then
-						if not HasPedGotWeapon(ped, 883325847) then
-							DrawText3Ds(stringCoords.x, stringCoords.y, stringCoords.z + 1.2, Config.Strings.PurchaseJerryCan)
-
-							if IsControlJustReleased(0, 38) then
-								TriggerServerEvent('QBCore:Server:AddItem', "weapon_petrolcan", 1)
-								TriggerEvent("inventory:client:ItemBox", QBCore.Shared.Items["weapon_petrolcan"], "add")
-								TriggerServerEvent('fuel:pay', Config.JerryCanCost, GetPlayerServerId(PlayerId()))
-							end
-						else
-							local refillCost = Round(Config.RefillCost * (1 - GetAmmoInPedWeapon(ped, 883325847) / 4500))
-
-							if refillCost > 0 then
-								if currentCash >= refillCost then
-									DrawText3Ds(stringCoords.x, stringCoords.y, stringCoords.z + 1.2, Config.Strings.RefillJerryCan .. refillCost)
-
-									if IsControlJustReleased(0, 38) then
-										TriggerServerEvent('fuel:pay', refillCost, GetPlayerServerId(PlayerId()))
-
-										SetPedAmmo(ped, 883325847, 4500)
+						DrawText3Ds(stringCoords.x, stringCoords.y, stringCoords.z + 1.2, Config.Strings.PurchaseJerryCan)
+						if IsControlJustReleased(0, 38) then
+							QBCore.Functions.TriggerCallback('fuel:server:hasJerryCans', function(hasCans, items)  
+								if hasCans then
+									if #items > 0 then
+										local can = items[1]
+					
+										local refillCost = Round(Config.RefillCost * (1 - can.info.ammo / 4500))
```

Comment:
You should be change this line to:
local refillCost = Round(Config.RefillCost * (1 - can.info.quality/ 4500))

Reply:
Thanks! Good spot!